src/status_im2/contexts/emoji_picker/view.cljs

```diff
@@ -0,0 +1,200 @@
+            [react-native.core :as rn]
+            [react-native.gesture :as gesture]
+            [react-native.platform :as platform]
+            [reagent.core :as reagent]
+            [status-im2.contexts.emoji-picker.constants :as constants]
+            [status-im2.contexts.emoji-picker.data :as emoji-picker.data]
+            [status-im2.contexts.emoji-picker.style :as style]
+            [status-im2.contexts.emoji-picker.utils :as emoji-picker.utils]
+            [utils.debounce :as debounce]
+            [utils.i18n :as i18n]
+            [utils.re-frame :as rf]
+            [utils.transforms :as transforms]))
+
+(defn- on-press-category
+  [id index active-category scroll-ref]
+  (reset! active-category id)
+  (some-> ^js @scroll-ref
+          (.scrollToIndex #js
+                           {:index    (emoji-picker.data/get-section-header-index-in-data index)
+                            :animated false})))
+
+(defn- handle-on-viewable-items-changed
+  [event active-category filtered-data]
+  (when-not @filtered-data
+    (let [viewable-item (-> (oops/oget event "viewableItems")
+                            transforms/js->clj
+                            first
+                            :item)
+          is-header?    (and (map? viewable-item) (:header? viewable-item))
+          section-key   (if is-header?
+                          (:id viewable-item)
+                          (emoji-picker.data/section-group (get-in (vec viewable-item) [0 :group])))]
+      (when (and (some? section-key) (not= @active-category section-key))
+        (reset! active-category section-key)))))
+
+(defn- get-item-layout
+  [_ index]
+  #js
+   {:length constants/get-item-height
+    :offset (* constants/get-item-height index)
+    :index  index})
+
+(defn- section-header
+  [{:keys [title]} _ _ {:keys [theme]}]
+  [quo/divider-label
+   {:tight?          false
+    :container-style (style/section-header theme)}
+   (i18n/label title)])
+
+(defn- emoji-item
+  [{:keys [unicode] :as emoji} col-index callback close]
+  (let [on-press          (fn []
+                            (when callback
+                              (callback unicode emoji))
+                            (close))
+        last-item-on-row? (= (inc col-index) constants/emojis-per-row)]
+    (fn []
+      [rn/pressable
+       {:style    (style/emoji-container last-item-on-row?)
+        :on-press on-press}
+       [rn/text
+        {:style                    {:font-size constants/emoji-size}
+         :adjusts-font-size-to-fit true
+         :allow-font-scaling       false}
+        unicode]])))
+
+(defn- emoji-row
+  [row-data _ _ {:keys [callback close]}]
+  (into [rn/view {:style style/emoji-row-container}]
+        (map-indexed
+         (fn [col-index {:keys [hexcode] :as emoji}]
+           ^{:key (str hexcode col-index)}
+           [emoji-item emoji col-index callback close])
+         row-data)))
+
+(defn- render-item
+  [item index separator render-data]
```

Comment:
is it possible to use a map here instead?

Reply:
Not without changing the implementation of `gesture/flat-list` and further `rn-flat-list/base-list-props` to call the render function passing a map.

Reply:
No, that's not possible. 

https://github.com/status-im/status-mobile/blob/2f13dfe22aaf33ec0644f14291f10128c8fabb5e/src/react_native/flat_list.cljs#L7-L13

As @ilmotta mentioned, It requires a refactor of FlatList usages across the app.